pkg/prowgen/jobbase.go

```diff
@@ -0,0 +1,185 @@
+package prowgen
```

Comment:
Most functional code in this file was moved here from `prowgen.go`, especially from the `GenerateJobs` method. It is just structured differently, allowing just parts of the generation code to be used only when needed.